genesis/example/member.go

```diff
@@ -17,134 +17,140 @@
+
 }
 
-// newMemberDomain creates new instance of MemberDomain.
-func newMemberDomain(parent object.Parent) (*memberDomain, error) {
+// newMember creates new instance of member.
+func newMember(parent object.Parent) (Member, error) {
 	if parent == nil {
 		return nil, fmt.Errorf("parent must not be nil")
 	}
-
-	return &memberDomain{
-		BaseDomain: *domain.NewBaseDomain(parent, MemberDomainName),
-	}, nil
+	m := &member{}
+	m.CompositeMap = make(map[string]object.Composite)
+	return m, nil
 }
 
-// GetClassID returns string representation of MemberDomain's class.
-func (md *memberDomain) GetClassID() string {
-	return class.MemberDomainID
+// GetClassID returns string representation of member's class.
+func (m *member) GetClassID() string {
+	return class.MemberID
 }
 
-// CreateMember creates new member as a child to domain storage.
-func (md *memberDomain) CreateMember(fc factory.Factory) (string, error) {
-	member, err := fc.Create(md)
-	if err != nil {
-		return "", err
-	}
-	if member == nil {
-		return "", fmt.Errorf("factory returns nil")
-	}
-
-	record, err := md.ChildStorage.Set(member)
-	if err != nil {
-		return "", err
-	}
-
-	return record, nil
+// GetUsername returns member's username.
+func (m *member) GetUsername() string {
+	return m.Username
 }
 
-// GetMember returns member from its record in domain storage.
-func (md *memberDomain) GetMember(record string) (resolver.Proxy, error) {
-	member, err := md.ChildStorage.Get(record)
-	if err != nil {
-		return nil, err
-	}
-
-	result, ok := member.(resolver.Proxy)
-	if !ok {
-		return nil, fmt.Errorf("object with record `%s` is not `Proxy` instance", record)
+// GetPublicKey returns member's public key.
+func (m *member) GetPublicKey() *rsa.PublicKey {
+	if m.privateKey == nil {
+		key := generateKey()
+		m.privateKey = key
 	}
 
-	return result, nil
+	return &m.privateKey.PublicKey
 }
 
-type memberDomainProxy struct {
+type memberProxy struct {
 	resolver.BaseProxy
 }
 
-// newMemberDomainProxy creates new proxy and associates it with new instance of MemberDomain.
-func newMemberDomainProxy(parent object.Parent) (*memberDomainProxy, error) {
-	instance, err := newMemberDomain(parent)
+// newMemberProxy creates new proxy and associates it with new instance of Member.
+func newMemberProxy(parent object.Parent) (*memberProxy, error) {
+	instance, err := newMember(parent)
 	if err != nil {
 		return nil, err
 	}
-	return &memberDomainProxy{
+	return &memberProxy{
 		BaseProxy: resolver.BaseProxy{
 			Instance: instance,
 		},
 	}, nil
 }
 
-// CreateMember is a proxy call for instance method.
-func (mdp *memberDomainProxy) CreateMember(fc factory.Factory) (string, error) {
-	return mdp.Instance.(MemberDomain).CreateMember(fc)
+// GetUsername is a proxy call for instance method.
+func (mp *memberProxy) GetUsername() string {
+	return mp.Instance.(Member).GetUsername()
+}
+
+// GetPublicKey is a proxy call for instance method.
+func (mp *memberProxy) GetPublicKey() *rsa.PublicKey {
```

Comment:
CreateComposite and GetComposite is not a public methods

Reply:
No, but for now we should implement all interface methods in proxy object

Reply:
I think I'll do this in my job